tutorials/tutorial-01.md

```diff
@@ -0,0 +1,117 @@
+# Tutorial 2: Tests for Distributions and samples
+
+For the following exercises and tutorials, you will be building up a codebase in your repository.
+
+During this tutorial, you will copy over some functions from Exercise sheet 1, and modify them to pass the tests that you can find on the bottom of this sheet.
+
+> [!TIP]
+> Reminder on how to run tests in [the setup instructions](https://github.com/RUB-EP1/ExercisesDataAnalysisWS2425/blob/main/exercises/setup.md#back-to-julia-running-tests).
+
+We will need some of the p.d.f.s in un-normalized form (such that the optimizations during fitting can work faster)
+
+1. Gaussian
+2. Polynomial
+3. Relativistic Breit-Wigner
+4. Voigt profile
+
+To pass the tests, their signatures should look like this:
+
+1. `gauss(x; μ, σ, a)`, where `a` is an arbitrary normalization factor
+2. `poly(x; coeffs)`, where `coeffs` is a tuple of coefficients. Hint: implement the polynomials in a loop.
+3. `rbw(x; M, Γ, a)`
+4. `voigt(x; M, Γ, σ, a)`
+
+We will also use the sampling methods and generalize them to take any function as input. Their signature looks like this:
+
+```julia
+function sample_rejection(f, n, support, nbins=1000)
+# f is the function,
+# n the sample size (passing the method),
+# support is a tuple specifying in which range the function will be sampled
+# nbins is the number of equidistant points in "support" for finding the maximum of the function (the default value is 1000)
+```
```

Comment:
Suggest placing here a docstring that students can reuse

(chatgpt)

```
"""
    sample_rejection(f, n, support; nbins=1000)

Generates `n` samples using the rejection sampling method for a given function `f` over a specified `support` range.

# Arguments
- `f::Function`: The function to sample from.
- `n::Int`: The number of samples to generate.
- `support::Tuple{T, T}`: A tuple specifying the range `(a, b)` where the function `f` will be sampled.
- `nbins::Int`: Optional. The number of equidistant points in `support` to find the maximum of `f`. Default is `1000`.

# Returns
- An array of `n` samples generated from the distribution defined by `f`.

# Example
```julia
data = sample_rejection(cos, 1000, (0, pi); nbins=1000)
```
"""
```